packages/daemon/src/service/trace-manager.ts

```diff
@@ -0,0 +1,250 @@
+import { Transform, TransformCallback } from 'stream';
+import { EventEmitter } from 'events';
+import { open, close, write } from 'fs-extra';
+import { provide, scope, ScopeEnum } from 'midway';
+import { StringDecoder } from 'string_decoder';
+import { generateId, PipelineStatus, PluginTypeI } from '@pipcook/pipcook-core';
+import Debug from 'debug';
+
+const debug = Debug('tracer');
+
+export type PipcookEventType = 'log' | 'job_status';
+
+/**
+ * base pipcook event, defined the fields `type` and `data`
+ */
+export class BasePipcookEvent {
+  data?: any;
+  constructor(public type: PipcookEventType) {}
+}
+
+/**
+ * pipcook event data type for job status change
+ */
+export class JobStatusChangeEvent extends BasePipcookEvent {
```

Comment:
Could this be defined at JobController?

Reply:
I think we should move them out finally, like the `JobResp`/`TraceResp`, because the sdk depends on them too.